bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -143,167 +146,266 @@ void extractInertial(
-        const auto &parentLinkName = joint->ParentName();
-        const auto *parent = model.LinkByName(parentLinkName);
-        const auto &childLinkName = joint->ChildName();
-        const auto *child = model.LinkByName(childLinkName);
+        gzerr << "Unable to find nested model " << nestedModelName << std::endl;
+        return errors;
+      }
 
-        switch (joint->Type())
-        {
-          case ::sdf::JointType::FIXED:
-          case ::sdf::JointType::REVOLUTE:
-          case ::sdf::JointType::PRISMATIC:
-          case ::sdf::JointType::BALL:
-            break;
-          default:
-            gzerr << "Joint type [" << (std::size_t)(joint->Type())
-                  << "] is not supported by "
-                  << "gz-physics-bullet-featherstone-plugin. "
-                  << "Replaced by a fixed joint.\n";
-        }
+      math::Pose3d p;
+      errors = nestedModel->SemanticPose().Resolve(p);
+      if (!errors.empty())
+        return errors;
+      _pose = _pose * p;
+      return linkPoseInModelTree(_pose, nestedLinkName, nestedModel);
+    }
+  }
+  else
+  {
+    const auto *link = _model->LinkByName(_linkName);
+    if (!link)
+    {
+      gzerr << "Unable to find link " << _linkName << std::endl;
+      return errors;
+    }
+    math::Pose3d p;
+    errors = link->SemanticPose().Resolve(p);
+    if (!errors.empty())
+      return errors;
 
-        if (child == parent)
-        {
-          gzerr << "The Link [" << parentLinkName << "] is being attached to "
-                << "itself by Joint [" << joint->Name() << "] in Model ["
-                << rootModelName << "]. That is not allowed.\n";
-          return false;
-        }
+    _pose = _pose * p;
+  }
+  return errors;
+}
 
-        if (nullptr == parent && parentLinkName != "world")
-        {
-          gzerr << "The link [" << parentLinkName << "] cannot be found in "
-                << "Model [" << rootModelName << "], but joint ["
-                << joint->Name() << "] wants to use it as its parent link\n";
-          return false;
-        }
-        else if (nullptr == parent)
-        {
-          // This link is attached to the world, making it the root
-          if (nullptr != rootLink)
-          {
-            // A root already exists for this model
-            gzerr << "Two root links were found for Model [" << rootModelName
-                  << "]: [" << rootLink->Name() << "] and [" << childLinkName
-                  << "], but gz-physics-bullet-featherstone-plugin only "
-                  << "supports one root per Model.\n";
-            return false;
-          }
+/////////////////////////////////////////////////
+/// \brief Get pose of joint relative to link
+/// \param[out] _resolvedPose Pose of joint relative to link
+/// \param[in] _model Parent model of joint
+/// \param[in] _joint Joint name
+/// \param[in] _link Scoped link name
+::sdf::Errors resolveJointPoseRelToLink(math::Pose3d &_resolvedPose,
```

Comment:
I think the logic in this function can be simplified. I've started trying to do so, but it's not working yet

Reply:
ok thanks for the tip. I actually tried it before but couldn't get it to work and now I found that it was because I wasn't passing the fully scoped name.  Code should be simpler now. [575e73b](https://github.com/gazebosim/gz-physics/pull/574/commits/575e73b587c4f4523bdc97b00be7423dd17391c5)